tests/data/test_dataset_access.py

```diff
@@ -168,11 +155,67 @@ def test_attributes_must_be_list(self, attributes, error_type, error_message):
             )
 
     @pytest.mark.parametrize(
-        ("data_type", "description", "attributes"),
+        ("description", "error_message"),
+        [
+            (
+                {"molname": [15], "basis": ["6-31G"], "bondlength": ["0.46"]},
+                "Invalid type 'int' for parameter 'molname'",
+            ),
+            (
+                {"molname": ["H2"], "basis": ["6-31G"], "bondlength": [1.1, 1 + 2j]},
+                r"Invalid bondlength '\(1\+2j\)'. Must be a string, int or float",
+            ),
+            (
+                {"layout": [1, [1, 4]]},
```

Comment:
Yes, they can specify the layout of the spin system's lattice.

Reply:
If they choose `qspin`, it becomes a required keyword parameter. Now, they can enter tuples or lists, instead of just strings (ie. they can say `(1,4)` instead of `"1x4"`)

Reply:
@obliviateandsurrender Is  "layout" a widely-used term from the literature for this property?

Reply:
I don't think I've seen a particular way to refer to this. Most papers I've seen investigate only one of these layouts.